player.py

```diff
@@ -202,6 +206,8 @@ def calc_miss_chance(self):
             (8. - miss_reduction) + (6.5 - dodge_reduction)
         )
         self.dodge_chance = 0.01 * (6.5 - dodge_reduction)
+        spell_miss_reduction = min(self.spell_hit_chance * 100, 17.0)
+        self.spell_miss_chance = 0.01 * (17.0 - spell_miss_reduction)
```

Comment:
What's missing here is that dynamic changes to Hit Rating during a fight (from trinket procs etc.) or more importantly, Hit Rating adjustments within `calc_stat_weights`, won't correctly propagate to the spell miss chance.